tests/bgp/test_bgp_router_id.py

```diff
@@ -0,0 +1,139 @@
+
+pytestmark = [
+    pytest.mark.topology('any')
+]
+
+logger = logging.getLogger(__name__)
+
+CUSTOMIZED_BGP_ROUTER_ID = "8.8.8.8"
+
+
+def verify_bgp(enum_asic_index, duthost, expected_bgp_router_id, neighbor_type, nbrhosts):
+    output = duthost.shell("show ip bgp summary", module_ignore_errors=True)["stdout"]
+
+    # Verify router id from DUT itself
+    pattern = r"BGP router identifier (\d+\.\d+\.\d+\.\d+)"
+    match = re.search(pattern, output)
+    pytest_assert(match, "Cannot get actual BGP router id from [{}]".format(output))
+    pytest_assert(match.group(1) == expected_bgp_router_id,
+                  "BGP router id unexpected, expected: {}, actual: {}".format(expected_bgp_router_id, match.group(1)))
+
+    # Verify BGP sessions are established
+    run_bgp_facts(duthost, enum_asic_index)
+
+    # Verify from peer device side to check
+    if neighbor_type not in ["sonic", "eos"]:
+        logger.warning("Unsupport neighbor type for neighbor bgp check: {}".format(neighbor_type))
+    local_ip_map = {}
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    for _, item in cfg_facts.get("BGP_NEIGHBOR", {}).items():
+        if "." in item["local_addr"]:
+            local_ip_map[item["name"]] = item["local_addr"]
+
+    for neighbor_name, nbrhost in nbrhosts.items():
+        pytest_assert(neighbor_name in local_ip_map, "Cannot find local ip for {}".format(neighbor_name))
+        if neighbor_type == "sonic":
+            cmd = "show ip neighbors {}".format(local_ip_map[neighbor_name])
+        elif neighbor_type == "eos":
+            cmd = "/usr/bin/Cli -c \"show ip bgp neighbors {}\"".format(local_ip_map[neighbor_name])
+        output = nbrhost["host"].shell(cmd, module_ignore_errors=True)['stdout']
+        pattern = r"BGP version 4, remote router ID (\d+\.\d+\.\d+\.\d+)"
+        match = re.search(pattern, output)
+        pytest_assert(match, "Cannot get remote BGP router id from [{}]".format(output))
+        pytest_assert(match.group(1) == expected_bgp_router_id,
+                      "BGP router id is unexpected, local: {}, fetch from remote: {}"
+                      .format(expected_bgp_router_id, match.group(1)))
+
+
+@pytest.fixture()
+def loopback_ip(duthosts, enum_frontend_dut_hostname):
+    duthost = duthosts[enum_frontend_dut_hostname]
+    cfg_facts = duthost.config_facts(host=duthost.hostname, source="running")['ansible_facts']
+    loopback_ip = None
+    loopback_table = cfg_facts.get("LOOPBACK_INTERFACE", {})
+    for key in loopback_table.get("Loopback0", {}).keys():
+        if "." in key:
+            loopback_ip = key.split("/")[0]
+    pytest_require(loopback_ip is not None, "Cannot get IPv4 address of Loopback0")
+    yield loopback_ip
+
+
+def restart_bgp(duthost):
+    duthost.reset_service("bgp")
+    duthost.restart_service("bgp")
+    pytest_assert(wait_until(100, 10, 10, duthost.is_service_fully_started_per_asic_or_host, "bgp"), "BGP not started.")
```

Comment:
Is this time enough for some of low performance platform? What is the common setting for config reload?